### ISSUE TITLE
Training is extremely case sensitive

### ISSUE THREAD

**Issue by jvence** [MahmoudWahdan/dialog-nlu#6]
Noticed something interesting but probably not related to your implementation. When you train the model, it seems to be VERY case-sensitive. So training the model with:
i love you -> LoveIntent
I hate you -> HateIntent

and then trying to predict (notice the lower-case 'i'):
i hate you -> yields prediction: LoveIntent

I suspect that this is inherent in the BERT & ALBERT model but thought I would bring it to your attention.

**Comment by MahmoudWahdan** [comment]
@jvence  The used BERT model `bert_uncased_L-12_H-768_A-12` is case-insensitive. The current ALBERT model `albert_base` supposed to be case-insensitive as well, because as you can see in the below code in `bert_vectorizer.py`
```
        vocab_file, do_lower_case = self.sess.run(
            [
                tokenization_info["vocab_file"],
                tokenization_info["do_lower_case"],
            ]
        )
```
you will find `do_lower_case` is `True` in both cases. However, after debugging I found that ALBERT tokenizer output maintain the case. So, I'll fix this, since I have access to their tokenizer.

If you were using BERT in your example, The problem you mentioned may be due to the model is under-fitted. So, please test the same the same sentence with multiple cases and you will find the same output with the same probability as well.
Example 1:
```
{
	"utterance": "make me a reservation in south carolina"
}
```
Example 2:
```
{
	"utterance": "Make me a reservation in South Carolina"
}
```

both example 1 and 2 have exactly the same response:
```
{
    "intent": {
        "confidence": "0.9966",
        "name": "BookRestaurant"
    },
    "slots": [
        {
            "end": 6,
            "slot": "state",
            "start": 5,
            "value": "South Carolina"
        }
    ]
}
```

**Comment by MahmoudWahdan** [comment]
@jvence Solved the case bug in `albert_tokenization.py` ec3e6c539b87c3ce7e9bb3c788010af44eda11cd

**Comment by jvence** [comment]
Thanks. Working perfectly.

**Comment by jvence** [comment]
Not sure if this is related but it it normal that the model doesn't recognize a certain sentence when a '?' or '.' is added to it?

**Comment by MahmoudWahdan** [comment]
@jvence Thank you for this important note.
Actually, philosophically, We don't need to do preprocessing for the text while using BERT or transformers in general. But, in our case the training data is indeed a preprocessed one :)
I did a search on the training set, I didn't find dots, commas, question marks, or any punctuation. 
I tried a query from the test set `where can i purchase the tv show time for heroes`. By adding dots or question mark directly after the last character `where can i purchase the tv show time for heroes?` or separated `where can i purchase the tv show time for heroes ?`, It really affects the `confidence` going from `0.9992` to `0.9989` to `0.999` respectively  and the slot `object_name` is detected as `time for heroes` to `time for heroes?` to `time for heroes ?` respectively.

I see that it depends on the data. punctuation shouldn't be removed, because it hold part of the meaning, but if this is the case in your training data, you should also remove it in prediction time by doing preprocessing before calling the API. 
What do you think? I really want to know your opinion.
